src/vs/workbench/parts/extensions/electron-browser/extensionsViews.ts

```diff
@@ -320,7 +320,9 @@ export class ExtensionsListView extends CollapsibleView {
 				}
 
 				return this.extensionsWorkbenchService.queryGallery(assign(options, { names, pageSize: names.length }))
-					.then(pager => new PagedModel(pager));
+					.then(pager => this.addRecommendationInfo(new PagedModel(pager || []), {
+						source: 'fileType'
+					}));
```

Comment:
Telemetry can be added from both directions... 

There will be some that should be set at the GalleryService level so that same gallery extension be it SearchedExtension, InstallingExtension, InstalledExtension has the same telemetry info

There will be some based on the View. We should be able to tell if an extension was installed from the Search view/Installed view/Popular view/Recommended view

And it just so happens that even inside the Recommended view, there can be different telemetry based on the source which cannot be deciphered from the search query